### DISCUSSION TITLE
Unable to modify ~/.zshrc config file

### DISCUSSION THREAD

**Discussion by IliaBadz** [ohmyzsh/ohmyzsh#10182] [Help] [answered]
I followed steps for installing zsh on manjaro but as soon as I try to enter command ~/.zshrc  I receive message that permission is denied. After using "chmod u+x  ~/.zshrc"  I typed ~/.zshrc again but ran into error:

/home/ilia/.oh-my-zsh/tools/check_for_upgrade.sh: line 32: syntax error near unexpected token `)' /home/ilia/.oh-my-zsh/tools/check_for_upgrade.sh: line 32: `() {' /home/ilia/.oh-my-zsh/oh-my-zsh.sh: line 21: autoload: command not found /home/ilia/.oh-my-zsh/oh-my-zsh.sh: line 39: syntax error near unexpected token `(' /home/ilia/.oh-my-zsh/oh-my-zsh.sh: line 39: `for plugin ($plugins); do'

I installed and uninstalled it several times. I checked if someone had similar issues. I tried to fix them as it was resolved before me but none of them worked. I am sure that I am not in bash anymore and that I switched to zsh. echo $SHELL shows "/usr/bin/zsh" . exec zsh did not help. Neither can I open it in vim. It returns that command is not found. 

Is there solution for this?

**Comment by jeffmhubbard** [comment, marked as answer]
You don't execute the config, you edit it. Also check the file name, in the title you write `~/.zshrc`, which is correct. In your explanation, you wrote `~/.zshcr`, which is incorrect.

`vim ~/.zshrc`

**Reply by IliaBadz** [reply]
Thanks for suggestion. My bad, sadly ~/.zshcr was typo only in question itself.

**Comment by robbyrussell** [comment]
@IliaBadz were you able to sort this out?

**Reply by IliaBadz** [reply]
Yes I did.

**Reply by rwmitchell** [reply]
How did you fix it?

**Reply by IliaBadz** [reply]
Basically my issue was that I tried to open ~/.zshrc file with "vi". I typed "vi ~/.zshrc" and even though file did open I could not make changes in it. And then went too far for searching answers for what could be wrong. Finally I opened file with "nano" text editor and was able to make necessary changes ( btw I did not need this command -> "chmod u+x ~/.zshrc" in my case). Hope this information is helpful.

**Reply by rwmitchell** [reply]
I don't think the actual solution was nano and ```chmod u+x ~/.zshrc``` would be wrong anyway.

~.zshrc needs to be owned and read/write for you.  Does not need to be executable.   Since you didn't look at the basic file info, it is hard to tell what happened.

**Reply by IliaBadz** [reply]
Could you please tell me what can I possibly do? How can I look at the basic file info?

**Reply by rwmitchell** [reply]
```ls -l ~/.zshrc```

usually works

**Reply by IliaBadz** [reply]
I found where that file was located (thanks for that), opened it and there were all edits that I made before. Shouldn't it be so?

**Reply by IliaBadz** [reply]
Everything works just fine as well.
